src/core/featuremodel.cpp

```diff
@@ -662,6 +662,34 @@ void FeatureModel::resetAttributes( bool partialReset )
   endResetModel();
 }
 
+bool FeatureModel::updateAttributesFromFeature( const QgsFeature &feature )
+{
+  bool updated = false;
+  const QgsFields fields = feature.fields();
+  for ( int i = 0; i < fields.size(); i++ )
+  {
+    const int idx = mFeature.fields().lookupField( fields[i].name() );
+    if ( idx >= 0 )
+    {
+      if ( mLayer && mLayer->primaryKeyAttributes().contains( idx ) )
+      {
+        continue;
+      }
+
+      if ( !mFeature.fields()[idx].defaultValueDefinition().expression().isEmpty() )
```

Comment:
```suggestion
      // We intentionally do not paste values on attributes that have default values
      if ( !mFeature.fields()[idx].defaultValueDefinition().expression().isEmpty() )
```

I guess this will become problematic one day, we recently discussed this on chat with you and Dave.